integrations/couchbase-document-store.md

```diff
@@ -0,0 +1,325 @@
+- `embedding` is also a property of the Document (just shown separately in the diagram for clarity) which is a vector of type `LIST[FLOAT]`.
+- `Search service` Where indexes specially purposed for Full Text Search and Vector search are created. The Search Service allows for efficient querying 
+and retrieval based on both text content and vector embeddings.
+
+`CouchbaseDocumentStore` requires the vector index to be created manually either by sdk or UI. Before writing documents you should make sure Documents are embedded by one of the provided [embedders](https://docs.haystack.deepset.ai/v2.0/docs/embedders). For example [SentenceTransformersDocumentEmbedder](https://docs.haystack.deepset.ai/v2.0/docs/sentencetransformersdocumentembedder) can be used in indexing pipeline to calculate document embeddings before writing those to Couchbase.
+
+## Installation
+
+`couchbase-haystack` can be installed as any other Python library, using pip:
+
+```bash
+pip install --upgrade pip # optional
+pip install sentence-transformers # required in order to run pipeline examples given below
+pip install couchbase-haystack
+```
+
+## Usage
+
+### Running Couchbase
+
+You will need a running instance of Couchbase to use the components from this package. There are several options available:
+
+- [Docker](https://docs.couchbase.com/server/current/getting-started/do-a-quick-install.html)
+- [Couchbase Cloud](https://www.couchbase.com/products/capella) - a fully managed cloud service
+- [Couchbase Server](https://www.couchbase.com/downloads) - installable on various operating systems
+
+The simplest way to start the database locally is with a Docker container:
+
+```bash
+docker run \
+    --restart always \
+    --publish=8091-8096:8091-8096 --publish=11210:11210 \
+    --env COUCHBASE_ADMINISTRATOR_USERNAME=admin \
+    --env COUCHBASE_ADMINISTRATOR_PASSWORD=passw0rd \
+    couchbase:enterprise-7.6.2
+```
+
+In this example, the container is started using Couchbase Server version `7.6.2`. The `COUCHBASE_ADMINISTRATOR_USERNAME` and `COUCHBASE_ADMINISTRATOR_PASSWORD` environment variables set the default credentials for authentication.
+
+> **Note:**  
+> Assuming you have a Docker container running, navigate to <http://localhost:8091> to open the Couchbase Web Console and explore your data.
+
+### Document Store
+
+Once you have the package installed and the database running, you can start using `CouchbaseDocumentStore` as any other document stores that support embeddings.
+
+```python
+from couchbase_haystack import CouchbaseDocumentStore
+
+document_store = CouchbaseDocumentStore(
+    cluster_connection_string= Secret.from_token("localhost"),
+    authenticator=CouchbasePasswordAuthenticator(
+        username = Secret.from_token("username"),
+        password = Secret.from_token("password")
+    ),
+    bucket = "haystack_bucket_name",
+    scope="haystack_scope_name",
+    collection="haystack_collection_name",
+    vector_search_index = "vector_search_index"
```

Comment:
Are all these parameters required? Also, is it straightforward to find the corresponding values on Couchbase?